toplevel/genprintval.ml

```diff
@@ -264,14 +264,14 @@ module Make(O : OBJ)(EVP : EVALPATH with type valu = O.t) = struct
         try
           find_printer depth env ty obj
         with Not_found ->
-          match get_desc ty with
-          | Tvar _ | Tunivar _ ->
+          match get_desc ty, get_desc (Ctype.expand_head_opt env ty) with
```

Comment:
That's looks like a good idea, but I would propose to move this expansion inside the `Tconstr` case. Also, it should be better to use the public version `expand_head` rather the optimisation-oriented `expand_head_opt`.

Reply:
Done